### ISSUE TITLE
Changing AMF_VIDEO_ENCODER_TARGET_BITRATE at runtime

### ISSUE THREAD

**Issue by maxest** [GPUOpen-LibrariesAndSDKs/AMF#294]
Every now and then I'd like to change bitrate of the stream. So I call:
`AMF_SET_PROPERTY( AMF_VIDEO_ENCODER_TARGET_BITRATE, targetBitrate );`
It seems to be working just fine. The thing that bothers me is that this call outputs this warning:
> Warning: static parameter set during encode state... 

Should I be worried about it? Is there a more appropriate way to change target bitrate?

**Comment by rhutsAMD** [comment]
Which OS, driver version, and GPU are you using?

**Comment by rhutsAMD** [comment]
Please try updating your driver to the latest version since the target bitrate parameter behavior has been updated.

 Are you setting any other parameters after initializing the encoder?

**Comment by maxest** [comment]
Thank you rhuts for your suggestion. It was spot-on indeed. Not only it resolved the warning but also I noticed bitrate to behave better. At least when I set AMF_VIDEO_ENCODER_TARGET_BITRATE and AMF_VIDEO_ENCODER_PEAK_BITRATE both to the same value. Surprisingly, when I set AMF_VIDEO_ENCODER_TARGET_BITRATE to 1 mbps, and AMF_VIDEO_ENCODER_PEAK_BITRATE to 20 mbps, the generated datsa produced ~20 mbps data, but quality was questionable. Anyway, setting both of these properties to a specific value gives stable and good results.

I now get a few other warnings, which seemingly do not cause any problems for me.
On call:
`if ( AMF_OK != gAMFInterface.GetFactory()->CreateComponent( gAMFInterface.GetContext(), AMFVideoEncoderVCE_AVC, &encoder ) ) {`
I get:
> 2022-02-09 19:29:24.709     449C [AMFEncoderCoreBaseImpl] Warning: Failed to QueryThroughput()!

Also, on call:
`if ( AMF_OK != encoder->Init( surfaceFormat, picInfo.width, picInfo.height ) ) {`
I get:

> 2022-02-09 19:33:26.493     449C [AMFEncoderCoreBaseImpl] Warning: Failed to QueryThroughput()!
> 2022-02-09 19:33:26.494     449C [AMFEncoderCoreBaseImpl] Warning: RequireThroughput() - Failed to require throughput.
